ledger-service/http-json/src/main/scala/com/digitalasset/http/json/DomainJsonDecoder.scala

```diff
@@ -40,28 +41,32 @@ class DomainJsonDecoder(
 
   def decodeCreateCommand(a: JsValue, jwt: Jwt, ledgerId: LedgerApiDomain.LedgerId)(implicit
       ev1: JsonReader[
-        domain.CreateCommand[JsValue, ContractTypeId.OptionalPkg]
-      ], // TODO #15098 .Template
+        domain.CreateCommand[JsValue, ContractTypeId.Template.OptionalPkg]
+      ],
       ec: ExecutionContext,
       lc: LoggingContextOf[InstanceUUID],
-  ): ET[domain.CreateCommand[lav1.value.Record, ContractTypeId.RequiredPkg]] = { // TODO #15098 .Template
+  ): ET[domain.CreateCommand[lav1.value.Record, ContractTypeId.Template.RequiredPkg]] = {
     val err = "DomainJsonDecoder_decodeCreateCommand"
     for {
       fj <- either(
         SprayJson
-          .decode[domain.CreateCommand[JsValue, ContractTypeId.OptionalPkg]](
-            a
-          ) // TODO #15098 .Template
+          .decode[domain.CreateCommand[JsValue, ContractTypeId.Template.OptionalPkg]](a)
           .liftErrS(err)(JsonError)
       )
 
-      tmplId <- templateId_(fj.templateId, jwt, ledgerId)
-
-      payloadT <- either(templateRecordType(tmplId))
+      tmplId <- contractTypeId_(fj.templateId, jwt, ledgerId)
+      resolvedTmplId <- either(
+        tmplId match {
+          case tId: ContractTypeId.Template.Resolved =>
+            \/-(tId: ContractTypeId.Template.Resolved)
+          case _ => -\/(JsonError("ContractTypeId has to be template id"))
+        }
+      )
```

Comment:
Did you not mean to just use `templateId_` here and so avoid the manual error checking?